core/store/src/trie/split_state.rs

```diff
@@ -35,21 +39,96 @@ impl Trie {
 }
 
 impl ShardTries {
+    pub fn apply_state_changes_to_split_states<'a>(
+        &self,
+        shard_uid: ShardUId,
+        state_root: StateRoot,
+        state_roots: &HashMap<ShardUId, StateRoot>,
+        changes: ConsolidatedStateChanges,
+        account_id_to_shard_id: &(dyn Fn(&AccountId) -> ShardUId + 'a),
+    ) -> Result<HashMap<ShardUId, TrieChanges>, StorageError> {
+        let trie = self.new_trie_update_view(shard_uid, state_root);
+        let mut trie_updates: HashMap<_, _> = self.get_trie_updates(state_roots);
+        let mut insert_receipts = Vec::new();
+        let mut delete_receipts = Vec::new();
+        for ConsolidatedStateChange { trie_key, value } in changes {
+            match trie_key {
+                TrieKey::DelayedReceiptIndices => {}
+                TrieKey::DelayedReceipt { index } => match value {
+                    Some(value) => {
+                        let receipt: Receipt = <_>::try_from_slice(&value).map_err(|_| {
```

Comment:
That's cute syntax, but I'd suggest simpler 
```suggestion
                        let receipt = Receipt::try_from_slice(&value).map_err(|_| {
```